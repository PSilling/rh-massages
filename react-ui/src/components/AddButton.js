// react imports
import React, { Component } from 'react';

/**
 * Custom add button component.
 */
class AddButton extends Component {

  render() {
    return(
      <span>
        <button type="button" className="btn" onClick={this.props.onAdd}>
          <span className="glyphicon glyphicon-plus"></span>
        </button>
      </span>
    );
  }
}

export default AddButton
