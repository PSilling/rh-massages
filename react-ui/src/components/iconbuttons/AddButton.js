// react imports
import React, { Component } from 'react';

// util imports
import _t from '../../util/Translations';

/**
 * Custom add button component.
 */
class AddButton extends Component {

  render() {
    return(
      <span>
        <button type="button" className="btn" onClick={this.props.onAdd} title={ _t.translate("Add") }>
          <span className="glyphicon glyphicon-plus"></span>
        </button>
      </span>
    );
  }
}

export default AddButton
