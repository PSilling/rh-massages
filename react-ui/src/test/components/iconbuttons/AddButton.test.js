// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import AddButton from '../../../components/iconbuttons/AddButton';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <AddButton onAdd={testFunction} />
        ),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-plus" }),
      treeJSON = testRenderer.toJSON();

  expect(testFunction).not.toHaveBeenCalled();

  button.props.onClick();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(button.props.onClick).toBe(testFunction);
  expect(treeJSON).toMatchSnapshot();
});
