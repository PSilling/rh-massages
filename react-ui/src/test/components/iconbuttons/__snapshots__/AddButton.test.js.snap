// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<span>
  <button
    className="btn btn-default"
    onClick={[Function]}
    title="PÅ™idat"
    type="button"
  >
    <span
      className="glyphicon glyphicon-plus"
    />
  </button>
</span>
`;
